client/components/platform-checkout/save-user/test/phone-number-input.test.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { render, screen, fireEvent } from '@testing-library/react';
+
+/**
+ * Internal dependencies
+ */
+import PhoneNumberInput from '../phone-number-input';
+
+describe( 'PhoneNumberInput', () => {
+	const handlePhoneNumberChangeMock = jest.fn();
+
+	it( 'should render phone number input', () => {
+		render(
+			<PhoneNumberInput
+				handlePhoneNumberChange={ handlePhoneNumberChangeMock }
+			/>
+		);
+		expect(
+			screen.queryByText( 'Mobile phone number' )
+		).toBeInTheDocument();
+	} );
+
+	it( 'should render the default selected country with code', () => {
+		render(
+			<PhoneNumberInput
+				handlePhoneNumberChange={ handlePhoneNumberChangeMock }
+			/>
+		);
+		expect(
+			screen.queryByRole( 'combobox', { name: 'United States: +1' } )
+		).toBeInTheDocument();
+	} );
+
+	it( 'should call the handlePhoneNumberChange with phone number including country code', () => {
+		render(
+			<PhoneNumberInput
+				handlePhoneNumberChange={ handlePhoneNumberChangeMock }
+			/>
+		);
+
+		expect( handlePhoneNumberChangeMock ).not.toHaveBeenCalled();
+
+		const input = screen.queryByPlaceholderText( '201-555-0123' ); // placeholder number for us
+		fireEvent.change( input, { target: { value: '201' } } );
+
+		expect( handlePhoneNumberChangeMock ).toHaveBeenCalledWith( '+1201' );
+	} );
+} );
```

Comment:
Probably need to mock the calls.

Reply:
Did you mean to mock the `intl-tel-input` package? I think this is fine; I think testing some of these visual features somewhat necessitates having the actual package present. That being said, I don't possess much mastery over these front-end tests, so I don't know how feasible mocking out something like that is in the first place. 🤷

Reply:
My comment was a bit unclear. Some tests were failing in the parent component tests because of `intl-tel-input` so I thought I would need to mock it there. But `npm install` fixed that actually. So we are good without the mock :smile: